erts/emulator/sys/unix/erl_child_setup.c

```diff
@@ -642,15 +646,29 @@ typedef struct exit_status {
 
 static Hash *forker_hash;
 
+/* Kill child process groups on VM termination so they don't become orphaned. */
+
+static void kill_child(pid_t os_pid) {
+    if (os_pid > 0 && kill(os_pid, SIGTERM) != 0) {
+        DEBUG_PRINT("error killing process %d: %d", os_pid, errno);
+    }
+}
+
+static void fun_kill_foreach(ErtsSysExitStatus *es, void *unused) {
+    kill_child(es->os_pid);
+}
+
+static void kill_all_children(void) {
+    DEBUG_PRINT("cleaning up by killing all %d child processes", forker_hash->nobjs);
+    hash_foreach(forker_hash, (HFOREACH_FUN)fun_kill_foreach, NULL);
+}
+
 static void add_os_pid_to_port_id_mapping(Eterm port_id, pid_t os_pid)
 {
-    if (port_id != THE_NON_VALUE) {
```

Comment:
This is the leak: the port_id actually will be empty, when open_port is called without exit_status.  Then the hash function places unrelated processes at the hashed location for 0.